source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -1314,17 +1384,63 @@ bool ServerContextImpl::isClientEcdsaCapable(const SSL_CLIENT_HELLO* ssl_client_
   return false;
 }
 
+OcspStapleAction ServerContextImpl::passesOcspPolicy(const ContextImpl::TlsContext& ctx) {
+  auto& response = ctx.ocsp_response_;
+  if (!Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.validate_ocsp_expiration_on_connection")) {
+    return response ? OcspStapleAction::Staple : OcspStapleAction::NoStaple;
+  }
+
+  if (ctx.is_must_staple_ && response && response->isExpired()) {
```

Comment:
What if `!response` in this case? Shouldn't that also fail? `if (ctx.is_must_staple_ && (!response || response->isExpired()))`